apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentService.java

```diff
@@ -60,7 +61,7 @@ public Mono<ExperimentPage> find(
         log.info("Finding experiments by '{}', page '{}', size '{}'", experimentSearchCriteria, page, size);
 
         if (experimentSearchCriteria.datasetDeleted()) {
-            return experimentDAO.findAllDatasetIds()
+            return experimentDAO.findAllDatasetIds(DatasetCriteria.builder().build())
```

Comment:
Shouldn't you add promptId here?

Reply:
Yes, but you use experimentDAO and query experiment table which has promptId, logic added in this PR, so promptId could be here.

Reply:
Added